### DISCUSSION TITLE
How to select all TreeView?

### DISCUSSION THREAD

**Discussion by lostluna1** [microsoft/microsoft-ui-xaml#10209] [Q&A] [answered]
Anyway, when I set IsSelected=true; I can only choose the last one

**Comment by lostluna1** [comment]
@ghost1372 Do you have any good ideas? Looking forward to your reply

**Reply by ghost1372** [reply]
Do you mean you want to select all treeview items? 
I think you should enable multiselect.
`SelectionMode="Multiple"`

**Reply by lostluna1** [reply]
I mean use code-behind to select all in TreeViewItem, not xaml "SelectionMode="Multiple""

**Reply by lostluna1** [reply]
My basic purpose is to bind the SelectedItem of the TreeViewItem according to the data in the vm, but since it is read-only, I cannot do so. I can only think of setting IsSelected of the TreeViewItem

**Comment by castorix** [comment, marked as answer]
When I do this in a test TreeView (tv1), it selects all items :

                tv1.SelectedItems.Clear();
                foreach (var rootNode in tv1.RootNodes)
                {
                    tv1.SelectedItems.Add(rootNode);
                }

![tv_selectall](https://github.com/user-attachments/assets/5badb95c-c711-4cc8-b0be-ba71539f0c41)

**Reply by lostluna1** [reply]
Your answer is perfectly, in china, about xaml info is so poor, very thanks your reply!!
